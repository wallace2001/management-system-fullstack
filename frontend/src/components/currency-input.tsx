'use client';

import { Input } from '@/components/ui/input';
import { useEffect, useState } from 'react';

type CurrencyInputProps = {
  value: number;
  onChange: (value: number) => void;
  placeholder?: string;
};

export function CurrencyInput({ value, onChange, placeholder }: CurrencyInputProps) {
  const [formatted, setFormatted] = useState('');

  useEffect(() => {
    const formattedValue = new Intl.NumberFormat('pt-BR', {
      style: 'currency',
      currency: 'BRL',
    }).format(value);
    setFormatted(formattedValue);
  }, [value]);

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const rawValue = e.target.value.replace(/\D/g, '');
    const numericValue = Number(rawValue) / 100;
    onChange(numericValue);
  };

  return (
    <Input
      inputMode="numeric"
      value={formatted}
      onChange={handleChange}
      placeholder={placeholder || 'R$ 0,00'}
    />
  );
}
