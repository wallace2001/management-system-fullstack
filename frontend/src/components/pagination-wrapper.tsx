'use client';

import {
  Pagination,
  PaginationContent,
  PaginationItem,
  PaginationPrevious,
  PaginationNext,
  PaginationLink,
  PaginationEllipsis,
} from '@/components/ui/pagination';

type Props = {
  currentPage: number;
  totalPages: number;
  onPageChange: (page: number) => void;
};

export function PaginationWrapper({
  currentPage,
  totalPages,
  onPageChange,
}: Props) {
  const pages = getPages(currentPage, totalPages);

  return (
    <Pagination className="mt-4">
      <PaginationContent>
        <PaginationItem>
          <PaginationPrevious
            onClick={() => currentPage > 1 && onPageChange(currentPage - 1)}
          />
        </PaginationItem>

        {pages.map((page, idx) => (
          <PaginationItem key={idx}>
            {page === '...' ? (
              <PaginationEllipsis />
            ) : (
              <PaginationLink
                isActive={page === currentPage}
                onClick={() => onPageChange(Number(page))}
              >
                {page}
              </PaginationLink>
            )}
          </PaginationItem>
        ))}

        <PaginationItem>
          <PaginationNext
            onClick={() =>
              currentPage < totalPages && onPageChange(currentPage + 1)
            }
          />
        </PaginationItem>
      </PaginationContent>
    </Pagination>
  );
}

function getPages(current: number, total: number): (number | string)[] {
  const delta = 2;
  const pages: (number | string)[] = [];

  for (let i = 1; i <= total; i++) {
    if (
      i === 1 ||
      i === total ||
      (i >= current - delta && i <= current + delta)
    ) {
      pages.push(i);
    } else if (pages[pages.length - 1] !== '...') {
      pages.push('...');
    }
  }

  return pages;
}
