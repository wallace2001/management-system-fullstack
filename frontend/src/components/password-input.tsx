'use client';

import { Input } from '@/components/ui/input';
import { cn } from '@/lib/utils';
import { Eye, EyeOff } from 'lucide-react';
import { useState } from 'react';

type PasswordInputProps = React.InputHTMLAttributes<HTMLInputElement>;

export function PasswordInput({ className, ...props }: PasswordInputProps) {
  const [showPassword, setShowPassword] = useState(false);

  return (
    <div className="relative">
      <Input
        type={showPassword ? 'text' : 'password'}
        className={cn('py-6 pr-10', className)}
        {...props}
      />
      <button
        type="button"
        onClick={() => setShowPassword((prev) => !prev)}
        className="absolute inset-y-0 right-3 flex items-center"
      >
        {showPassword ? (
          <EyeOff className="h-5 w-5 text-gray-600" />
        ) : (
          <Eye className="h-5 w-5 text-gray-600" />
        )}
      </button>
    </div>
  );
}
