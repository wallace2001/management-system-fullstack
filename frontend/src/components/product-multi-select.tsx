/* eslint-disable react-hooks/exhaustive-deps */
'use client';

import { cn } from '@/lib/utils';
import {
  Select,
  SelectContent,
  SelectItem,
  SelectTrigger,
  SelectValue,
} from '@/components/ui/select';
import { Input } from '@/components/ui/input';
import { useEffect, useState } from 'react';
import { Product } from '@/modules/shared/types/product';
import { useProductsQuery } from '@/modules/products/hooks/use-get-products';

type ProductMultiSelectProps = {
  value: Record<string, number>;
  onChange: (value: Record<string, number>) => void;
} & React.HTMLAttributes<HTMLDivElement>;

export function ProductMultiSelect({
  value,
  onChange,
  className,
  ...props
}: ProductMultiSelectProps) {
  const { data: products = [] } = useProductsQuery();
  const [selectedId, setSelectedId] = useState('');
  const [quantity, setQuantity] = useState(1);

  useEffect(() => {
    if (!value || typeof value !== 'object') {
      onChange({});
    }
  }, []);

  function handleAddProduct() {
    if (!selectedId || quantity < 1) return;

    const updated = {
      ...value,
      [selectedId]: (value[selectedId] || 0) + quantity,
    };

    onChange(updated);
    setSelectedId('');
    setQuantity(1);
  }

  function handleRemoveProduct(productId: string) {
    const updated = { ...value };
    delete updated[productId];
    onChange(updated);
  }

  return (
    <div className={cn('space-y-2', className)} {...props}>
      <div className="flex gap-2 items-center">
        <Select value={selectedId} onValueChange={setSelectedId}>
          <SelectTrigger className="w-full">
            <SelectValue placeholder="Selecione um produto" />
          </SelectTrigger>
          <SelectContent>
            {products.map((product: Product) => (
              <SelectItem key={product.id} value={product.id}>
                {product.name}
              </SelectItem>
            ))}
          </SelectContent>
        </Select>

        <Input
          type="number"
          min={1}
          value={quantity}
          onChange={(e) => setQuantity(Number(e.target.value))}
          className="w-24"
          placeholder="Qtd"
        />

        <button
          type="button"
          onClick={handleAddProduct}
          className="px-4 py-2 bg-primary text-white text-sm rounded"
        >
          Adicionar
        </button>
      </div>

      <div className="text-sm text-muted-foreground space-y-1">
        {Object.entries(value).length > 0 ? (
          Object.entries(value).map(([id, qty]) => {
            const name = products.find((p) => p.id === id)?.name || id;
            return (
              <div
                key={id}
                className="flex items-center justify-between border px-2 py-1 rounded"
              >
                <span>
                  {name} â€” <strong>{qty}x</strong>
                </span>
                <button
                  type="button"
                  className="text-xs text-red-500 hover:underline"
                  onClick={() => handleRemoveProduct(id)}
                >
                  Remover
                </button>
              </div>
            );
          })
        ) : (
          <p>Nenhum produto adicionado.</p>
        )}
      </div>
    </div>
  );
}
