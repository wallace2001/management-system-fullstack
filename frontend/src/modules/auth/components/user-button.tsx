'use client';

import { Crown, LogOut } from 'lucide-react';

import { Avatar, AvatarFallback, AvatarImage } from '@/components/ui/avatar';
import {
  DropdownMenu,
  DropdownMenuContent,
  DropdownMenuItem,
  DropdownMenuSeparator,
  DropdownMenuTrigger,
} from '@/components/ui/dropdown-menu';
import { logout } from '@/lib/auth';

export const UserButton = () => {
  return (
    <DropdownMenu modal={false}>
      <DropdownMenuTrigger className="relative cursor-pointer outline-none">
        <div className="absolute -top-1 -left-1 z-10 flex items-center justify-center">
          <div className="flex items-center justify-center rounded-full bg-white p-1 drop-shadow-sm">
            <Crown className="size-3 fill-yellow-500 text-yellow-500" />
          </div>
        </div>
        <Avatar className="hover:opcaity-75 transition">
          <AvatarImage alt="" src="https://github.com/shadcn.png" />
          <AvatarFallback className="flex items-center justify-center bg-blue-500 font-medium text-white" />
        </Avatar>
      </DropdownMenuTrigger>
      <DropdownMenuContent align="end" className="w-60">
        <DropdownMenuSeparator />
        <DropdownMenuItem
          className="h-10"
          onClick={async () => {
            logout();
          }}
        >
          <LogOut className="mr-2 size-4" />
          Log out
        </DropdownMenuItem>
      </DropdownMenuContent>
    </DropdownMenu>
  );
};
