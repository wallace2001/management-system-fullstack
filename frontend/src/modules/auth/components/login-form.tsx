'use client';

import { zodResolver } from '@hookform/resolvers/zod';
import { useForm } from 'react-hook-form';
import { Input } from '@/components/ui/input';
import {
  Form,
  FormItem,
  FormControl,
  FormLabel,
  FormMessage,
  FormField,
} from '@/components/ui/form';
import { PasswordInput } from '@/components/password-input';
import { LoginFormValues, loginSchema } from '../schemas/login-schema';
import { SubmitButton } from '@/components/submit-button';
import Link from 'next/link';
import { useLoginMutation } from '../hooks/use-login-mutation';

export function LoginForm() {
  const loginMutation = useLoginMutation();
  const form = useForm<LoginFormValues>({
    resolver: zodResolver(loginSchema),
    defaultValues: {
      username: '',
      password: '',
    },
  });

  const onSubmit = (values: LoginFormValues) => {
    loginMutation.mutate(values);
  };

  return (
    <Form {...form}>
      <form
        onSubmit={form.handleSubmit(onSubmit)}
        className="space-y-6 bg-gray-50 px-6 py-8 sm:px-12"
      >
        <FormField
          control={form.control}
          name="username"
          render={({ field }) => (
            <FormItem>
              <FormLabel>Usuário</FormLabel>
              <FormControl>
                <Input className="h-12" placeholder="Seu usuário" {...field} />
              </FormControl>
              <FormMessage />
            </FormItem>
          )}
        />

        <FormField
          control={form.control}
          name="password"
          render={({ field }) => (
            <FormItem>
              <FormLabel>Senha</FormLabel>
              <FormControl>
                <PasswordInput placeholder="Sua senha" {...field} />
              </FormControl>
              <FormMessage />
            </FormItem>
          )}
        />

        <SubmitButton>Entrar</SubmitButton>
        <p className="text-center text-sm text-gray-600">
          {'Ainda não tem uma conta? '}
          <Link href="/auth/register" className="font-semibold text-gray-800">
            Cadastrar
          </Link>
        </p>
      </form>
    </Form>
  );
}
