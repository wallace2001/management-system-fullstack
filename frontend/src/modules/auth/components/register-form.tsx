'use client';

import { zodResolver } from '@hookform/resolvers/zod';
import { useForm } from 'react-hook-form';
import { Input } from '@/components/ui/input';
import {
  Form,
  FormItem,
  FormControl,
  FormLabel,
  FormMessage,
  FormField,
} from '@/components/ui/form';
import { PasswordInput } from '@/components/password-input';
import { SubmitButton } from '@/components/submit-button';
import {
  Select,
  SelectTrigger,
  SelectValue,
  SelectContent,
  SelectItem,
} from '@/components/ui/select';
import Link from 'next/link';
import { RegisterFormValues, registerSchema } from '../schemas/register-schema';
import { useRegisterMutation } from '../hooks/use-register-mutation';

export function RegisterForm() {
  const registerMutation = useRegisterMutation();
  const form = useForm<RegisterFormValues>({
    resolver: zodResolver(registerSchema),
    defaultValues: {
      username: '',
      password: '',
      confirmPassword: '',
      role: 'USER',
    },
  });

  const onSubmit = (values: RegisterFormValues) => {
    registerMutation.mutate(values);
  };

  return (
    <Form {...form}>
      <form
        onSubmit={form.handleSubmit(onSubmit)}
        className="space-y-6 bg-gray-50 px-6 py-8 sm:px-12"
      >
        <FormField
          control={form.control}
          name="username"
          render={({ field }) => (
            <FormItem>
              <FormLabel>Usuário</FormLabel>
              <FormControl>
                <Input className="h-12" placeholder="Seu usuário" {...field} />
              </FormControl>
              <FormMessage />
            </FormItem>
          )}
        />

        <FormField
          control={form.control}
          name="password"
          render={({ field }) => (
            <FormItem>
              <FormLabel>Senha</FormLabel>
              <FormControl>
                <PasswordInput placeholder="Sua senha" {...field} />
              </FormControl>
              <FormMessage />
            </FormItem>
          )}
        />

        <FormField
          control={form.control}
          name="confirmPassword"
          render={({ field }) => (
            <FormItem>
              <FormLabel>Confirmar Senha</FormLabel>
              <FormControl>
                <PasswordInput placeholder="Confirme sua senha" {...field} />
              </FormControl>
              <FormMessage />
            </FormItem>
          )}
        />

        <FormField
          control={form.control}
          name="role"
          render={({ field }) => (
            <FormItem>
              <FormLabel>Role</FormLabel>
              <FormControl>
                <Select
                  defaultValue={field.value}
                  onValueChange={field.onChange}
                >
                  <SelectTrigger>
                    <SelectValue />
                  </SelectTrigger>
                  <SelectContent>
                    <SelectItem value="USER">USER</SelectItem>
                    <SelectItem value="ADMIN">ADMIN</SelectItem>
                  </SelectContent>
                </Select>
              </FormControl>
              <FormMessage />
            </FormItem>
          )}
        />

        <SubmitButton>Cadastrar</SubmitButton>
        <p className="text-center text-sm text-gray-600">
          {'Já tem uma conta? '}
          <Link href="/auth/login" className="font-semibold text-gray-800">
            Entrar
          </Link>
        </p>
      </form>
    </Form>
  );
}
