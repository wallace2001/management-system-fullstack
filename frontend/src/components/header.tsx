'use client';

import { usePathname } from 'next/navigation';
import { Button } from '@/components/ui/button';
import { Sheet, SheetContent, SheetTitle, SheetTrigger } from '@/components/ui/sheet';
import { UserButton } from '@/modules/auth/components/user-button';
import { Menu } from 'lucide-react';
import { User } from 'next-auth';
import Link from 'next/link';

export default function Header({ user }: { user: User }) {
  const pathname = usePathname();

  return (
    <header className="bg-background fixed top-0 z-50 w-full border-b px-4 sm:px-8">
      <div className="mx-auto flex h-20 max-w-screen-xl items-center justify-between">
        <p className="text-2xl font-bold">
          <Link href="/">Management</Link>
        </p>

        <div className="hidden items-center gap-2 md:flex">
          <Button
            variant={pathname === '/products' ? 'default' : 'outline'}
            asChild
          >
            <Link href="/products">Produtos</Link>
          </Button>

          <Button
            variant={pathname === '/orders' ? 'default' : 'outline'}
            asChild
          >
            <Link href="/orders">Pedidos</Link>
          </Button>

          {user?.id && (
            <Button
              variant={pathname === '/users' ? 'default' : 'outline'}
              asChild
            >
              <Link href="/users">Usuários</Link>
            </Button>
          )}

          {user?.id ? <UserButton /> : (
            <Button asChild>
              <Link href="/auth">Entrar</Link>
            </Button>
          )}
        </div>

        <div className="md:hidden">
          <Sheet>
            <SheetTrigger asChild>
              <Button variant="outline" size="icon">
                <Menu className="h-5 w-5" />
              </Button>
            </SheetTrigger>
            <SheetContent side="right" className="w-64 p-6 space-y-4">
              <SheetTitle />
              <div className="space-y-2 pt-10">
                <Button
                  variant={pathname === '/products' ? 'default' : 'ghost'}
                  className="w-full justify-start"
                  asChild
                >
                  <Link href="/products">Produtos</Link>
                </Button>

                <Button
                  variant={pathname === '/orders' ? 'default' : 'ghost'}
                  className="w-full justify-start"
                  asChild
                >
                  <Link href="/orders">Pedidos</Link>
                </Button>

                {user?.id && (
                  <Button
                    variant={pathname === '/users' ? 'default' : 'ghost'}
                    className="w-full justify-start"
                    asChild
                  >
                    <Link href="/users">Usuários</Link>
                  </Button>
                )}

                {user?.id ? (
                  <div className="pt-4">
                    <UserButton />
                  </div>
                ) : (
                  <Button variant="default" className="w-full" asChild>
                    <Link href="/auth">Entrar</Link>
                  </Button>
                )}
              </div>
            </SheetContent>
          </Sheet>
        </div>
      </div>
    </header>
  );
}
