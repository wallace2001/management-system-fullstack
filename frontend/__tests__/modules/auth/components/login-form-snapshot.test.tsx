import { render } from '@testing-library/react';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { LoginForm } from '@/modules/auth/components/login-form';

jest.mock('@/modules/auth/hooks/use-login-mutation', () => ({
  useLoginMutation: () => ({
    mutate: jest.fn(),
    isLoading: false,
  }),
}));

describe('LoginForm Snapshot', () => {
  it('renderiza corretamente', () => {
    const queryClient = new QueryClient();

    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <LoginForm />
      </QueryClientProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
