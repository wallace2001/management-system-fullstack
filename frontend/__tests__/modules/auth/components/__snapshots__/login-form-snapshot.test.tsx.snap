// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Snapshot renderiza corretamente 1`] = `
<div>
  <form
    class="space-y-6 bg-gray-50 px-6 py-8 sm:px-12"
  >
    <div
      class="grid gap-2"
      data-slot="form-item"
    >
      <label
        class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
        data-error="false"
        data-slot="form-label"
        for=":r0:-form-item"
      >
        Usuário
      </label>
      <input
        aria-describedby=":r0:-form-item-description"
        aria-invalid="false"
        class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex w-full min-w-0 rounded-md border bg-transparent px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive h-12"
        data-slot="form-control"
        id=":r0:-form-item"
        name="username"
        placeholder="Seu usuário"
        value=""
      />
    </div>
    <div
      class="grid gap-2"
      data-slot="form-item"
    >
      <label
        class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
        data-error="false"
        data-slot="form-label"
        for=":r1:-form-item"
      >
        Senha
      </label>
      <div
        class="relative"
      >
        <input
          aria-describedby=":r1:-form-item-description"
          aria-invalid="false"
          class="file:text-foreground placeholder:text-muted-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 border-input flex h-9 w-full min-w-0 rounded-md border bg-transparent px-3 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive py-6 pr-10"
          data-slot="form-control"
          id=":r1:-form-item"
          name="password"
          placeholder="Sua senha"
          type="password"
          value=""
        />
        <button
          class="absolute inset-y-0 right-3 flex items-center"
          type="button"
        >
          <svg
            class="lucide lucide-eye h-5 w-5 text-gray-600"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2.062 12.348a1 1 0 0 1 0-.696 10.75 10.75 0 0 1 19.876 0 1 1 0 0 1 0 .696 10.75 10.75 0 0 1-19.876 0"
            />
            <circle
              cx="12"
              cy="12"
              r="3"
            />
          </svg>
        </button>
      </div>
    </div>
    <button
      class="focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive shrink-0 cursor-pointer gap-2 rounded-md font-medium whitespace-nowrap transition-all outline-none focus-visible:ring-[3px] disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4 bg-primary text-primary-foreground hover:bg-primary/90 shadow-xs px-4 py-2 has-[>svg]:px-3 flex h-10 w-full items-center justify-center text-sm"
      data-slot="button"
      type="submit"
    >
      Entrar
      <span
        aria-live="polite"
        class="sr-only"
        role="status"
      >
        Submit form
      </span>
    </button>
    <p
      class="text-center text-sm text-gray-600"
    >
      Ainda não tem uma conta? 
      <a
        class="font-semibold text-gray-800"
        href="/auth/register"
      >
        Cadastrar
      </a>
    </p>
  </form>
</div>
`;
